app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -235,7 +474,43 @@ export default Controller.extend(CanCheckEmails, {
     },
 
     merge(targetUsername) {
-      return this.model.merge({ targetUsername });
+      const user = this.model;
+      const location = document.location.pathname;
+
+      const bootboxDiv = bootbox.dialog(I18n.t("admin.user.merging_user"));
+      let formData = { context: location };
+
+      if (targetUsername) {
+        formData["target_username"] = targetUsername;
+      }
+
+      this.model
+        .merge(formData)
+        .then((response) => {
+          if (response.success) {
+            messageBus.subscribe("/merge_user", (data) => {
+              if (data.merged) {
+                if (/^\/admin\/users\/list\//.test(location)) {
+                  DiscourseURL.redirectTo(location);
+                } else {
+                  DiscourseURL.redirectTo(
+                    `/admin/users/${data.user.id}/${data.user.username}`
+                  );
+                }
+              } else if (data.message) {
+                bootboxDiv.find(".modal-body").html(data.message);
```

Comment:
Also here, is there a way we could use regular ember for updating the body?

Reply:
This is tricky. I checked the potential options. We can create new `bootbox.dialog` and hide old one on each update but that will add more elements to the DOM. I guess that the best approach would be to render {{ d-modal }} to display and update progress instead of bootbox. Does it sound right to you?

Reply:
Oh I see. Yes that is a pain. Maybe as a follow up we could convert those to more Ember friendly updates (via dmodal or similar).